homeassistant/components/sensor/integration.py

```diff
@@ -0,0 +1,183 @@
+            self._unit_template = "{}{}{}".format(
+                "" if unit_prefix is None else unit_prefix,
+                "{}",
+                unit_time)
+            # we postpone the definition of unit_of_measurement to later
+            self._unit_of_measurement = None
+        else:
+            self._unit_of_measurement = unit_of_measurement
+
+        self._unit_prefix = UNIT_PREFIXES[unit_prefix]
+        self._unit_time = UNIT_TIME[unit_time]
+
+    async def async_added_to_hass(self):
+        """Handle entity which will be added."""
+        await super().async_added_to_hass()
+        state = await self.async_get_last_state()
+        if state:
+            try:
+                self._state = Decimal(state.state)
+            except ValueError as err:
+                _LOGGER.warning("Could not restore last state: %s", err)
+
+        @callback
+        def calc_integration(entity, old_state, new_state):
+            """Handle the sensor state changes."""
+            if old_state is None or\
+                    old_state.state in [STATE_UNKNOWN, STATE_UNAVAILABLE] or\
+                    new_state.state in [STATE_UNKNOWN, STATE_UNAVAILABLE]:
+                return
+
+            if self._unit_of_measurement is None:
+                unit = new_state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
+                self._unit_of_measurement = self._unit_template.format(
+                    "" if unit is None else unit)
+
+            try:
+                # integration as the Riemann integral of previous measures.
+                elapsed_time = (new_state.last_updated
+                                - old_state.last_updated).total_seconds()
+                area = (Decimal(new_state.state)
+                        + Decimal(old_state.state))*Decimal(elapsed_time)/2
+                integral = area / (self._unit_prefix * self._unit_time)
+
+                assert isinstance(integral, Decimal)
+            except ValueError as err:
+                _LOGGER.warning("While calculating integration: %s", err)
+            except DecimalException as err:
+                _LOGGER.warning("Invalid state (%s > %s): %s",
+                                old_state.state, new_state.state, err)
+            except AssertionError as err:
+                _LOGGER.error("Could not calculate integral: %s", err)
+            else:
+                self._state += integral
+                self.async_schedule_update_ha_state()
+
+        async_track_state_change(
+            self.hass, self._sensor_source_id, calc_integration)
+
+        @callback
+        def async_set_state(entity, old_state, new_state):
+            """Handle set state from dev-state."""
+            try:
+                self._state = Decimal(new_state.state)
+            except ValueError:
+                _LOGGER.error("State must be a number")
+
+        async_track_state_change(
+            self.hass, self.entity_id, async_set_state)
```

Comment:
Why are we tracking the state of this entity?

Reply:
Let me refer to previous comment: https://github.com/home-assistant/home-assistant/pull/19703#discussion_r245418577

I can delete it, but that was something my beta testers expected to work.

Reply:
I don't think we should do this. The dev tools are for developers or for debugging.

Reply:
done